transport/src/main/java/io/camunda/zeebe/transport/stream/impl/RemoteStreamServiceImpl.java

```diff
@@ -28,14 +29,20 @@ public class RemoteStreamServiceImpl<M extends BufferReader, P extends BufferWri
   private final RemoteStreamerImpl<M, P> streamer;
   private final RemoteStreamEndpoint<M> apiServer;
   private final RemoteStreamRegistry<M> registry;
+  private final RemoteStreamRestarter<P> remoteStreamRestarter;
+  private final ClusterMembershipService clusterMembershipService;
 
   public RemoteStreamServiceImpl(
       final RemoteStreamerImpl<M, P> streamer,
       final RemoteStreamEndpoint<M> apiServer,
-      final RemoteStreamRegistry<M> registry) {
+      final RemoteStreamRegistry<M> registry,
+      final RemoteStreamRestarter<P> remoteStreamRestarter,
+      final ClusterMembershipService clusterMembershipService) {
```

Comment:
@npepinpe I am not sure about passing the clusterMembershipService all the way down here, should we pass the member id instead? Or perhaps there is a more elegant way? wdyt?